qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,308 @@
+        # Check if target is the same for guest and host
+        if host_qemu["target"] != guest_qemu["target"]:
+            test.error(
+                "Guest target target '{}' differs from host '{}'".format(
+                    guest_qemu['target'],
+                    host_qemu['target']
+                )
+            )
+        if "module_id" in host_qemu and "module_id" in guest_qemu:
+            # Check if module_id is newer than the installed one
+            if int(host_qemu["module_id"]) > int(guest_qemu["module_id"]):
+                test.log.info(
+                    "Host module '{}' is newer than guest one '{}'".format(
+                        host_qemu['module_id'],
+                        guest_qemu['module_id']
+                    )
+                )
+            elif int(host_qemu["module_id"]) == int(guest_qemu["module_id"]):
+                test.cancel(
+                    "Host module '{}' is the same like guest one '{}'".format(
+                        host_qemu['module_id'],
+                        guest_qemu['module_id']
+                    )
+                )
+            else:
+                test.cancel(
+                    "Host module '{}' is older than guest one '{}'".format(
+                        host_qemu['module_id'],
+                        guest_qemu['module_id']
+                    )
+                )
+        # Check if qemu-version in the available and guest module differs
+        qemu_version_compare = _compare_qemu_versions(
+            host_qemu["nvr"],
+            guest_qemu["nvr"]
+            )
+        if qemu_version_compare[0]:
+            test.log.info(
+                "Available qemu-kvm '{}' is newer compared to guest's '{}'".format(
+                    *qemu_version_compare[1:]
+                )
+            )
+        elif not qemu_version_compare[0]:
+            test.cancel(
+                "Available qemu-kvm '{}' is older or same compared to guest's '{}'".format(
+                    *qemu_version_compare[1:]
+                )
+            )
+        if "module_id" in host_qemu:
+            return host_qemu['module_id']
+        else:
+            return host_qemu['nvr']
+
+    def update_qemu(session, install_id):
+        """
+        Prepare repository containing the newest version of qemu-kvm package,
+        handle modules if needed and run upgrade to the newest version
+        """
+        def nice_format(ugly_format):
+            better_format = ugly_format.replace("  ", " ")
+            if "  " in better_format:
+                return nice_format(better_format)
+            else:
+                return better_format
+        build_or_module = "--install-brewpkg"
+        if install_id.isnumeric():
+            build_or_module = "--install-virtqemu"
+        # Prepare module or build repo containing newer version of qemu-kvm
+        cmd = f"python3 \
+            /root/envutils/software_management/component_management.py \
```

Comment:
I prefer to specify the target path when clone it in L53 and use it here.

Reply:
Yes it will be better. Changed.